test/TestProjects/MgmtCustomizations/Customization/PetStoreData.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using Azure.Core;
+
+namespace MgmtCustomizations
+{
+    [CodeGenSerialization(nameof(Id), SerializationValueHook = nameof(WriteId), DeserializationValueHook = nameof(ReadId))]
+    public partial class PetStoreData
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal void WriteId(Utf8JsonWriter writer)
```

Comment:
If this is not in use, can just remove it?

Reply:
it is used, in the attribute.
This project does not enable the mrw feature, therefore we did not see the write method to call this, but if we enable that, this will have usages.